src/routes/process-user-vocabulary/components/AudioRecordOverlay.tsx

```diff
@@ -0,0 +1,205 @@
+const cleanUpMeteringResults = (meteringResults: number[]): number[] => {
+  const filteredResults = meteringResults.filter(el => el !== Math.min(...meteringResults))
+  return filteredResults.map(el => el + Math.abs(Math.min(...filteredResults)))
+}
+
+const getCurrentMetering = (metering?: number): number => {
+  if (!metering) {
+    return 0
+  }
+  return Platform.select({
+    ios: metering,
+    android: metering * androidFactor,
+  }) as number
+}
+
+const AudioRecordOverlay = ({
+  onClose,
+  setShowAudioRecordOverlay,
+  audioRecorderPlayer,
+  setRecordingPath,
+  recordingPath,
+  setRecordSeconds,
+}: AudioRecordOverlayProps): ReactElement => {
+  const [meteringResults, setMeteringResults] = useState<number[]>([])
+  const [recordingTime, setRecordingTime] = useState<string>(recordingTimeInit)
+  const [isPressed, setIsPressed] = useState<boolean>(false)
+  const { permissionGranted, permissionRequested } = useGrantPermissions(
+    Platform.OS === 'ios' ? PERMISSIONS.IOS.MICROPHONE : PERMISSIONS.ANDROID.RECORD_AUDIO
+  )
+  const { hold, talk } = getLabels().general.audio
+  const { description } = getLabels().general.audio.noAuthorization
+  const theme = useTheme()
+
+  useEffect(() => {
+    if (!recordingPath) {
+      setMeteringResults([])
+      setRecordSeconds(0)
+    }
+  }, [recordingPath, setRecordSeconds])
+
+  const cleanedMetering = useMemo(() => cleanUpMeteringResults(meteringResults), [meteringResults])
+
+  const onStopRecording = async (): Promise<void> => {
+    await audioRecorderPlayer.stopRecorder()
+    audioRecorderPlayer.removeRecordBackListener()
+    setShowAudioRecordOverlay(false)
+  }
+
+  const onStartRecording = async (): Promise<void> => {
+    setMeteringResults([])
+    const uri = await audioRecorderPlayer.startRecorder(undefined, undefined, true)
+    audioRecorderPlayer.addRecordBackListener(async e => {
+      setMeteringResults(oldMeteringResults => [
+        ...oldMeteringResults,
+        Math.floor(getCurrentMetering(e.currentMetering)),
+      ])
+      setRecordingTime(audioRecorderPlayer.mmss(Math.floor(e.currentPosition / factor)))
+      setRecordSeconds(e.currentPosition)
+      if (e.currentPosition > maxRecordingTime) {
+        await onStopRecording()
+      }
+    })
+    setRecordingPath(uri)
+  }
+
+  return (
+    <RNModal visible transparent animationType='fade' onRequestClose={() => onClose()}>
+      <Container>
+        <Icon onPress={() => onClose()}>
+          <CloseIcon width={theme.spacingsPlain.lg} height={theme.spacingsPlain.lg} />
+        </Icon>
+        <>
+          {permissionGranted && (
+            <Content>
+              <HeadingContainer>
+                <Heading>{isPressed ? talk : hold}</Heading>
+              </HeadingContainer>
+              <InfoContainer>
+                <MeteringInfo>
+                  {cleanedMetering.map(element => (
+                    <MeteringBar key={uuid.v4().toString()} height={element} />
```

Comment:
what is your suggestion in this case. index shouldn't be used and timestamps don't work here

Reply:
From the react docs (https://reactjs.org/docs/lists-and-keys.html#keys):
> When you don’t have stable IDs for rendered items, you may use the item index as a key as a last resort

and
> We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state.

I think since the order of items should not change we should be good with just indexes. Especially since with your solution `uuid.v4()` is run on every render for every element anew, therefore changes each time and therefore is even worse performance wise (didn't test this so not that sure if that is actually the case).

Reply:
ok perfect. i just revert that to index